shotgun_api3/lib/mockgun/mockgun.py

```diff
@@ -656,11 +657,6 @@ def _get_field_from_row(self, entity_type, row, field):
                 elif not isinstance(field_value, dict):
                     raise ShotgunError("Invalid deep query field %s.%s" % (entity_type, field))
 
-                # make sure that types in the query match type in the linked field
-                if entity_type2 != field_value["type"]:
```

Comment:
You can check the test_find_by_sub_entity_field I've added in my previous commit.
It seem that this error get triggered when filtering with multi-entities. This work on a real shotgun server but fail in mockgun.
A simple example is the 'entity' field on a 'Task'.
sg.find('Task', [['entity.Asset.code', 'is', 999]])
and
sg.find('Task', [['entity.Shot.code', 'is', 999]])
Are valid queries.

If check if important for something else, I might be able to extend it to work better with multi entities by querying what are the valid entity types.